getDataTableReader.html

```diff
@@ -0,0 +1,154 @@
+
+        let pageNumber = '${pageSelector.val()}';
+
+        //  update the page data upon a selector change.
+        pageSelector.change(function () {
+          getDataAsync(pageSelector.val())
+        });
+
+        // default values if null.
+        if (pageSelector.val() === null) {
+          pageNumber = 0
+        };
+
+        // Get the first page of data
+        getDataButton.onclick = async () => {
+          getDataAsync(pageNumber)
+        }
+      }
+
+      // Get the underlying data using the DataTableReader
+      async function getDataAsync (pageNumber) {
+        // Get data from a specified sheet
+        const sheet = viz.workbook.activeSheet.worksheets.find(
+          sheet => sheet.name === 'Storm Map Sheet'
+        );
+        const datasource = await sheet.getDataSourcesAsync();
+
+        // Get the logical tables from the first data source
+        const logicalTables = await datasource[0].getLogicalTablesAsync();
+
+        // Loop through each table that was used in creating this data source
+        logicalTables.forEach(function (table) {
+          console.log(`Name of logical table: ${table.caption}`);
+          document.getElementById(
+            'info'
+          ).innerHTML = `<p>Name of logical table: ${table.caption}</p>`
+        });
+
+        // Get the data from the first logical table
+        const logicalTableData = await datasource[0].getLogicalTableDataAsync(
+          logicalTables[0].id
+        );
+        console.log(logicalTableData);
+        console.log(`Total row count: ${logicalTableData.totalRowCount}`);
+        // Display the row count
+        document
+          .getElementById('info')
+          .insertAdjacentHTML(
+            'beforeend',
+            `<p>Total row count: ${logicalTableData.totalRowCount}</p>`
+          );
+
+        // Set the page size to 1000 for this sample; the default value is 10000 rows
+        const pageRowCount = 1000;
+        // Create the data table reader
+        const pageReader = await datasource[0].getLogicalTableDataReaderAsync(
+          logicalTables[0].id,
+          pageRowCount
+        );
+        // Display the number of pages to read
+        console.log(`Number of pages: ${pageReader.pageCount}`);
+        document
+          .getElementById('info')
+          .insertAdjacentHTML(
+            'beforeend',
+            `<p>Number of pages: ${pageReader.pageCount}</p>`
+          );
+
+        // Get the page data based on page number requested
+        let currentPageDataTable = await pageReader.getPageAsync(pageNumber);
+        console.log(currentPageDataTable);
+
+
+        // Add data to HTML element.
+        const target = document.getElementById('dataTarget')
+        target.innerHTML = `<h4>Data Table Page ${pageNumber} Data:</h4><p>${JSON.stringify(
+          currentPageDataTable.data
+        )}</p>`;
+
+        // Free up memory when done
+        await pageReader.releaseAsync();
```

Comment:
Let's provide a good pattern for developers to follow by putting this in a finally block. That means that after we create the pageReader, the remainder needs to be in a try block.

Reply:
Great. Thanks for that suggestion.  I tried to add it. Let me know if you catch any errors.